### DISCUSSION TITLE
WatchFragment method in the GraphQLCache class

### DISCUSSION THREAD

**Discussion by ReziaBanks** [zino-app/graphql-flutter#1422] [Ideas]
I'm currently working on a flutter project that requires infinite scroll pagination basically in every single page. I decided to use the [infinite_scroll_pagination](https://pub.dev/packages/infinite_scroll_pagination) package as it makes the entire development easy.

Using the package with the graphql package has been an issue because readFragment function doesn't return Stream.

```
Widget _buildPostSliverListView() {
  return PagedSliverList<int, Fragment$Post>.separated(
      pagingController: _pagingController,
      separatorBuilder: (context, index) => Gap(16),
      builderDelegate: PagedChildBuilderDelegate(
        itemBuilder: (context, post, index) {
          return AppPostCard(post);
        },
      ),
  );
}
```

The issue with the code above, is that it won't react to writeFragment changes. Eg (user liked a post)
With the addition of a watchFragment method I can wrap the AppPostCard in a StreamBuilder and that would solve the issue.

**Comment by ReziaBanks** [comment]
The ferry package has the exact method I'm talking about -> https://pub.dev/documentation/ferry/latest/ferry/Cache/watchFragment.html

**Comment by Woutwo** [comment]
I also run into this. Within the PagedSliverList it would be nice to just watch a few fragments instead of having to re-render the entire list all the time
